New APIG application resoruce supported

Lance52259 opened the pull request:
<!-- Thanks for sending a pull request! -->

**What this PR does / why we need it**:
An application is an identity for accessing an API and can call the APIs to which it has been authorized.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
reference: #1221 
fixes: #1198 

**Special notes for your reviewer**:

**Release note**:
<!--  Steps to write your release note:
1. Use the release-note-* labels to set the release note state (if you have access)
2. Enter your extended release note in the below block; leaving it blank means using the PR title as the release note. If no release note is required, just write `NONE`.
-->

```release-note
1. New application resource support up to five application codes to be bound.
2. New acc test for application.
3. Document supported.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed
Basic test
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccApigApplicationV2_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccApigApplicationV2_basic -timeout 360m -parallel 4
=== RUN   TestAccApigApplicationV2_basic
=== PAUSE TestAccApigApplicationV2_basic
=== CONT  TestAccApigApplicationV2_basic
--- PASS: TestAccApigApplicationV2_basic (479.69s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud  479.755s
```

Lance52259 committed:
feat: new application resoruce of apig service

ShiChangkuo commented on huaweicloud/resource_huaweicloud_apig_application.go:
```diff
@@ -0,0 +1,300 @@
+			},
+			"instance_id": {
+				Type:     schema.TypeString,
+				Required: true,
```
**ForceNew** is missing

Lance52259 replied:
The `ForceNew` has been added.

ShiChangkuo commented on huaweicloud/resource_huaweicloud_apig_application_test.go:
```diff
@@ -0,0 +1,137 @@
+	"github.com/huaweicloud/golangsdk/openstack/apigw/v2/applications"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils/fmtp"
```
we can use fmt package in test files directlly.

Lance52259 replied:
The fmt has been roll back.

Lance52259 committed:
update the resource, test and document

- fix the wrong description in the document
- roll back fmtp updates
- Fix the problem that the code exceeds 120 characters
